### PR TITLE
Upgrade terraform-provider-azuread to v2.38.0

### LINKED ISSUE #363: Upgrade terraform-provider-azuread to v2.38.0
Release details: https://github.com/hashicorp/terraform-provider-azuread/releases/tag/v2.38.0

### PR DESCRIPTION
**Pull request by iwahbe**
This PR was generated via `$ upgrade-provider pulumi-azuread`.

---

Upgrading terraform-provider-azuread from 2.37.2 to 2.38.0.
Upgrading pulumi-terraform-bridge from v3.41.0 to v3.45.3.

Fixes #363

### CODE REVIEW

**Review by aq17 — APPROVED** [review]

**Comment by iwahbe** [comment]
Note: the breaking change detected by schema-tools was caused by https://github.com/hashicorp/terraform-provider-azuread/commit/54eeec3c357590820dbe2d148d6608a01cb02bea. I resolved it by pinning `MaxItemsOne` for the relevant field to `false` in `bridge-metadata.json`.

**Comment by thomas11** [comment]
Some data sources were removed in favor of auto-mapping while others remain - what determines that?

**Comment by iwahbe** [comment]
> Some data sources were removed in favor of auto-mapping while others remain - what determines that?

I generally remove a couple of resources or databases each time I manually touch `resources.go`. This serves as a check that `ComputeDefaults` was applied correctly.

File: sdk/python/pulumi_azuread/get_application_published_app_ids.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def get_application_published_app_ids(opts: Optional[pulumi.InvokeOptions] = Non
     """
     Use this data source to discover application IDs for APIs published by Microsoft.
 
-    This data source uses an [unofficial source of application IDs](https://github.com/manicminer/hamilton/blob/main/environments/published.go), as there is currently no available official indexed source for applications or APIs published by Microsoft.
```
Comment (kpitzen) [inline sdk/python/pulumi_azuread/get_application_published_app_ids.py:62]: Curious where this change came from.

**Review by kpitzen — APPROVED** [review]
Approved with one Q!

File: sdk/python/pulumi_azuread/get_application_published_app_ids.py (hunk 1 above)
Comment (iwahbe) [reply sdk/python/pulumi_azuread/get_application_published_app_ids.py:62]: https://github.com/hashicorp/terraform-provider-azuread/commit/e503e815c27330cb99bc4f42741242be8e293a66 introduced a forbidden link in that sentence, so the bridge expunged the sentence.

**Pull request merged**